.circleci/config.yml

```diff
@@ -58,21 +59,56 @@ jobs:
           name: "Check for broken links in the docs"
           command: |
             poetry run sphinx-build -b linkcheck docs docs/_build/linkcheck
-      
+
+      # Build docs and push to gh-pages.
+      # Note: git remote set-url is for setting a git url instead of an https
+      # one, which is needed so Shippable can be authenticated to push changes.
+      - run:
+          name: "Build docs and push to gh-pages"
+          command: >-
+            git config --global user.email "api-documentation@gro-intelligence.com" &&
+            git config --global user.name "Gro Intelligence" &&
+            git remote set-url origin git@github.com:$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME.git &&
+            poetry run bin/sphinx_push_ghpages.sh
+
       # Note: We need to install versioning library outside of the poetry venv.
-      # See note in CONTRIBUTING.md
-      # TODO: Build docs and push to gh-pages.
+      # See note in CONTRIBUTING.md.
       - run:
           name: "Build package"
           command: >-
             pip install poetry-dynamic-versioning &&
             poetry build
 
+      # Publish new development updates to TestPyPI whenever changes are merged
+      # to development.
+      # We only want to publish a new package when PRs are actually accepted and
+      # merged, which is why we check $CIRCLE_PULL_REQUEST.
+      - run:
+          name: "Push to testpypi"
+          command: >-
+            if [ "$CIRCLE_BRANCH" == "development" ] && [ -z $CIRCLE_PULL_REQUEST ]; then
+              poetry config repositories.testpypi https://test.pypi.org/legacy/ &&
+              poetry publish -u __token__ -p $TESTPYPI_TOKEN -r testpypi
+            fi
+
+      # Publish new releases to PyPI.
+      # We only check whether this is a tagged build and if so, publish to pypi
+      # Check that $CIRCLE_TAG is not a null string
+      - run:
+          name: "Push to pypi"
+          command: >-
+            if [ -n $CIRCLE_TAG ]; then
```

Comment:
This should be quoted:

`[ -n "$CIRCLE_TAG" ]`

Reply:
Cool thanks!

Reply:
> It looks like CircleCI tried to run the PyPI publishing step. That shouldn't have happened because this isn't a tagged release 
> 
> https://app.circleci.com/pipelines/github/kmetallic04/api-client/20/workflows/4eb13c5b-9f20-4114-936d-2ca6a625d4b8/jobs/24/parallel-runs/0/steps/0-113

Looks like the build is no longer attempting to upload the artifacts to pypi! Double quotes ftw